abdm_integrator/hip/tests/test_care_contexts.py

```diff
@@ -0,0 +1,191 @@
+from unittest.mock import patch
+
+import requests
+from django.contrib.auth.models import User
+from rest_framework.authtoken.models import Token
+from rest_framework.reverse import reverse
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST
+from rest_framework.test import APIClient, APITestCase
+
+from abdm_integrator.const import LinkRequestInitiator, LinkRequestStatus
+from abdm_integrator.exceptions import (
+    ERROR_CODE_INVALID,
+    ERROR_CODE_REQUIRED,
+    ERROR_CODE_REQUIRED_MESSAGE,
+    STANDARD_ERRORS,
+)
+from abdm_integrator.hip.exceptions import HIPError
+from abdm_integrator.hip.models import HIPLinkRequest, LinkCareContext, LinkRequestDetails
+from abdm_integrator.tests.utils import APITestHelperMixin
+from abdm_integrator.utils import ABDMCache
+
+
+class TestHIPLinkCareContextAPI(APITestCase, APITestHelperMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = User.objects.create_superuser(username='test_user', password='test')
+        cls.token = Token.objects.create(user=cls.user)
+        cls.link_care_context_url = reverse('link_care_context')
+
+    def setUp(self):
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.token}')
+
+    def tearDown(self):
+        LinkCareContext.objects.all().delete()
+        HIPLinkRequest.objects.all().delete()
+        LinkRequestDetails.objects.all().delete()
+
+    @classmethod
+    def tearDownClass(cls):
+        User.objects.all().delete()
+        super().tearDownClass()
+
+    @staticmethod
+    def _link_care_context_sample_request_data():
+        return {
+            'accessToken': 'abcdefghi',
+            'hip_id': '1001',
+            'patient': {
+                'referenceNumber': 'Test_001',
+                'display': 'Test User',
+                'careContexts': [
+                    {
+                        'referenceNumber': str(uuid.uuid4()),
+                        'display': 'Made a Test Visit',
+                        'hiTypes': [
+                            'Prescription', 'WellnessRecord'
+                        ]
+                    }
+                ]
+            }
+        }
+
+    @staticmethod
+    def _insert_link_request(request_data, user, gateway_request_id):
+        link_request_details = LinkRequestDetails.objects.create(
+           hip_id=request_data['hip_id'],
+           patient_reference=request_data['patient']['referenceNumber'],
+           patient_display=request_data['patient']['display'],
+           initiated_by=LinkRequestInitiator.HIP,
+           status=LinkRequestStatus.SUCCESS
+        )
+        HIPLinkRequest.objects.create(user=user, gateway_request_id=gateway_request_id,
+                                      link_request_details=link_request_details)
+        for care_context in request_data['patient']['careContexts']:
+            LinkCareContext.objects.create(reference=care_context['referenceNumber'],
```

Comment:
Why not use https://docs.djangoproject.com/en/stable/ref/models/querysets/#bulk-create

Reply:
Addressed in https://github.com/dimagi/abdm-python-integrator/pull/4/commits/2f7e3ed89adcf6d423b4e7a49be2a87bd8c0ad18